text/0000-modifiers.md

```diff
@@ -0,0 +1,404 @@
+      eventOptions
+    ];
+    this.element.addEventListener(eventType, callback, eventOptions);
+  }
+}
+```
+
+#### `didUpdate` semantics
+
+`didUpdate` is called whenever any of the parameters used by the modifier are updated. `didUpdate` has the same signature as `didInsertElement`.
+
+This hook has the following timing semantics:
+
+**Always**
+- called **after** the arguments to the modifier have changed
+
+**Never**
+- called if the arguments to the modifier are constants
+
+Below is an example of how this hook could be used:
+
+```js
+import Modifier from '@ember/modifier';
+
+export default class extends Modifier {
+  this.listenerOptions = undefined;
+
+  willInsertElement() {
+    this.element.style = 'background-color: green;'
+  }
+
+  didInsertElement([ eventType, callback ], eventOptions) {
+    this.listenerOptions = [
+      eventType,
+      callBack,
+      eventOptions
+    ];
+    this.element.addEventListener(eventType, callback, eventOptions);
+  },
+
+  didUpdate([ eventType, callback ], eventOptions) {
+    this.element.removeEventListener(...this.listenerOptions);
+    this.element.addEventListener(eventType, callback, eventOptions);
+    this.listenerOptions = [eventType, callback, eventOptions];
+  }
+}
+```
+
+#### `willDestroyElement` semantics
+
+`willDestroyElement` is called during the destruction of the element. It receives no arguments.
+
+Below is an example of how this hook could be used:
+
+```js
+import Modifier from '@ember/modifier';
+
+export default class extends Modifier {
+  this.listenerOptions = undefined;
+
+  willInsertElement() {
+    this.element.style = 'background-color: green;'
+  }
+
+  didInsertElement([ eventType, callback ], eventOptions) {
+    this.listenerOptions = [
+      eventType,
+      callBack,
+      eventOptions
+    ];
+    this.element.addEventListener(eventType, callback, eventOptions);
+  },
+
+  didUpdate([ eventType, callback ], eventOptions) {
+    this.element.removeEventListener(...this.listenerOptions);
+    this.element.addEventListener(eventType, callback, eventOptions);
+    this.listenerOptions = [eventType, callback, eventOptions];
+  }
+
+  willDestroyElement() {
+    this.element.removeEventListener(...this.listenerOptions);
+  }
+}
+```
+
+This hook has the following timing semantics:
+
+**Always**
+- called **after** all children modifier's `willDestroyElement` hook is called
+- called **after** all children component's `willDestroyComponent` hook is called.
+
+ **May or May Not**
+- be called in the same tick as DOM removal
+
+### Relationship With Components
+
+Element modifers are intended to be directly installed onto HTML elements and can not be invoked on a component invocation. They also cannot be passed as arguments to a component. The examples below will result in a compile error.
+
+```hbs
+<FooBar {{draggable x=x y=y}} />
+```
+
+```hbs
+<FooBar (draggable x=x y=y) />
+```
+
+This also means that they have no relationship to `...attributes`.
```

Comment:
If I understand this, that means that a developer would not be able to forward the modifier. If so:
- What are the drawbacks of forwarding a modifier?
- What does that means for the `{{on}}` modifier? (currently can be forwarded)
  - Not giving modifiers the ability to forward and not changing `{{on}}` creates an inconsistent API
  - Changing `{{on}}` would be a breaking change

Thoughts?

Reply:
I think this is simply outdated information, as modifiers _are_ "forwarded" in `...attributes`. The RFC for that was created and merged _after_ this RFC was last updated: https://github.com/emberjs/rfcs/pull/435